controllers/databaseclusterbackup_controller.go

```diff
@@ -736,6 +736,19 @@ func (r *DatabaseClusterBackupReconciler) reconcilePG(
 		if controllerutil.RemoveFinalizer(backup, everestv1alpha1.DBBackupStorageProtectionFinalizer) {
 			return true, r.Update(ctx, backup)
 		}
+
+		// For the scheduled backups, need to requeue reconciliation one more time
+		// because the circular OwnerReferences dependency between dbb and pg-backup
+		// along with the foreground deletion (which gets propagated to the owned resources)
+		// may make the deletion stuck. The backward OwnerReference "pg-backup owns dbb" can be
+		// safely deleted since the dbb already has the deletion timestamp.
+		// The problem doesn't show up for pxc and psmdb bc the additional reconciliation requeue happens
+		// when the storage cleanup finalizer is deleted from the upstream backup.
```

Comment:
I'm still a bit confused why this worked for PSMDB and PXC.. even in the additional requeue we did there, we never explicitly removed the OwnerRefs. So why doesn't that lead to a deadlock? Does the `foregroundDeletion` finalizer not get propagated there? 🤔

Reply:
You're right, we never delete the `OwnerReference` from `pxc` and `psmdb`, the thing is that we only need _one more requeue_, and for that we need to change something in the CR so we could rely on the change as an indicator. It is safe to delete the `OwnerReference` from the `dbb` because the OwnerReference used there only to delete dbb in case the retention copies number is exceeded, but since the dbb already has the deletion timestamp in place, it's safe to delete the reference. 

We can also rely on some other change to be able to make the requeue only once. Removing the OwnerReference is just one option which seems to be the safest one. 

another option would be instead of the new block that this PR introduces just to add
```
return true, nil
``` 
so it would work like this: if the `dbb` has the deletion timestamp in place, the reconciliation will be requeued endless until the CR is deleted. It adds an unnecessary overhead and in general such endless loops can go out of control easily.

Reply:
On a second thought, let me try to remove the circular dependency in a separate PR. I'm going to remove the direction "upstream backup owns everest backup" and let a watcher track the upstream deletion so that the related dbb would be deleted too.

Reply:
Created the PR to remove circular OwnerReferences https://github.com/percona/everest-operator/pull/419 

Moving the current PR back to draft and will close it once we accept the OwnerReferences removing